tiledb/api/src/context.rs

```diff
@@ -46,8 +48,21 @@ impl Drop for RawContext {
     }
 }
 
-pub trait ContextBound<'ctx> {
-    fn context(&self) -> &'ctx Context;
+// I tried using this to mark all the types that just return a self.context
+// vs the types that return self.member.context(). It all compiled until I
+// tried implementing CApiInterface trait for it which caused a trait
+// implementation conflict I couldn't figure out. I'm leaving this and all
+// the marker traits commented out just so we have something to search for
```

Comment:
To be honest I'm kind of surprised that this made it as far as it did.
I'm sure the compiler would eventually have complained that `Self` doesn't necessarily have a `context` field.  It's kind of a form of specialization.

That is one downside with the compilation errors coming in stages, you can work really hard clearing out the errors from one stage only to have the next come around and give you another 100 errors

Reply:
Oooh, I need to delete that. The entire point for going this direction in the first place was to guarantee that we returned called clone() when getting down to a `ContextBound` implementation that has an actual `Context` member. But given that the trait now returns `Context` instead of `&Context` it's a compiler error to not return `self.context.clone()` which is all I wanted.

Reply:
> to not return self.context.clone() which is all I wanted

I think the recommendation is to qualify it a la `Rc::clone(&self.context)` which is functionally not different but tells the reader that it's not an expensive clone.  I don't remember where I read that.

Reply:
Huh. Apparently it comes from the Rust book and there's a (disabled due to false positive) lint for this approach. Good enough for me to change to that style.